metadata/src/main/java/org/apache/kafka/controller/BrokersToElrs.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.controller;
```

Comment:
To help review this file, you can refer to the BrokersToIsrs.java
The reasons not to modify the BrokersToIsrs are:
1. The BrokersToIsrs is complicated, hard to extend.
2. You may think that we can simply let the BrokersToIsrs treat the ISR and ELR the same, storing the ISR and ELR together. However, the ISR and ELR do not behave exactly the same(fence a broker will only remove it from ISRs).

Reply:
I guess the alternative would be to extend the BrokersToIsrs to include a ELR_FLAG similar to the LEADER_FLAG. I agree this would probably be complex to implement. If I recall from the KIP, the ELR is normally empty, so maybe having this parallel data structure isn't so bad (since it will normally not be populated)?

@cmccabe since you worked on `BrokersToIsrs` originally, do you have any opinion here?

Reply:
Updated the comment in the ReplicationControlManager.

Reply:
If we keeping this as a second data structure and we need to ensure BrokersToIsrs and BrokersToElrs are disjoint, I think we need some guards/checks. For example, it seems we are always updating both data structures

```
brokersToIsrs.update(record.topicId(), record.partitionId(), prevPartInfo.isr,
  newPartInfo.isr, prevPartInfo.leader, newPartInfo.leader);
brokersToElrs.update(record.topicId(), record.partitionId(), prevPartInfo.elr,
  newPartInfo.elr);
```

If we need to ensure a replica is not in the ISR and ELR for correctness, we should have a check.